pkg/clusterapi/system.go

```diff
@@ -0,0 +1,357 @@
+package clusterapi
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"text/template"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
+
+	"github.com/openshift/installer/data"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/clusterapi/internal/process"
+	"github.com/openshift/installer/pkg/clusterapi/internal/process/addr"
+	"github.com/openshift/installer/pkg/types/aws"
+	"github.com/openshift/installer/pkg/types/azure"
+	"github.com/openshift/installer/pkg/types/gcp"
+	"github.com/openshift/installer/pkg/types/ibmcloud"
+	"github.com/openshift/installer/pkg/types/nutanix"
+	"github.com/openshift/installer/pkg/types/vsphere"
+)
+
+var (
+	wg          = &sync.WaitGroup{}
+	ctx, cancel = context.WithCancel(signals.SetupSignalHandler())
+	once        = &sync.Once{}
+)
+
+// Teardown shuts down the local capi control plane and all its controllers.
+func Teardown() {
+	once.Do(func() {
+		cancel()
+		logrus.Info("Shutting down local Cluster API control plane...")
+		ch := make(chan struct{})
+		go func() {
+			wg.Wait()
+			close(ch)
+		}()
+		select {
+		case <-ch:
+			logrus.Info("Local control plane has completed operations")
+		case <-time.After(30 * time.Second):
+			logrus.Warn("Timed out waiting for local control plane to shut down")
+		}
+	})
+}
+
+// System creates a local capi control plane
+// to use as a management cluster.
+// TODO: Add support for existing management cluster.
+type System struct {
+	manifestDir string
+	lcp         *localControlPlane
+	Client      client.Client
+}
+
+// Name returns the human-friendly name of the asset.
+func (c *System) Name() string {
+	return "Cluster API System"
+}
+
+// Run launches the cluster-api system.
+func (c *System) Run(clusterID *installconfig.ClusterID, installConfig *installconfig.InstallConfig) (err error) {
```

Comment:
`clusterID` is unused (except to pass to the localcontrolplane.Run where it is also unused)

Reply:
Good catch, removed it